src/ziggurat/metrics.clj

```diff
@@ -2,43 +2,60 @@
+           (io.dropwizard.metrics5 MetricRegistry Meter MetricName Histogram)
+           (java.util.concurrent TimeUnit)))
 
 (defonce ^:private group (atom nil))
 
 (defonce metrics-registry
   (MetricRegistry.))
 
 (defn mk-meter
-  [category metric]
-  (let [metric-name (MetricRegistry/name ^String @group ^"[Ljava.lang.String;" (into-array String [category metric]))
-        tagged-metric (.tagged ^MetricName metric-name ^"[Ljava.lang.String;" (into-array String ["actor" @group]))]
-    (.meter ^MetricRegistry metrics-registry ^MetricName tagged-metric)))
+  ([category metric]
+   (mk-meter category metric nil))
+  ([category metric topic-entity-name]
+   (let [metric-name   (MetricRegistry/name ^String @group ^"[Ljava.lang.String;" (into-array String [category metric]))
+         tags          (if (nil? topic-entity-name)
+                         ["actor" @group]
+                         ["actor" @group "topic_name" topic-entity-name])
+         tagged-metric (.tagged ^MetricName metric-name ^"[Ljava.lang.String;" (into-array String tags))]
+     (.meter ^MetricRegistry metrics-registry ^MetricName tagged-metric))))
 
 (defn mk-histogram
-  [category metric]
-  (let [metric-name (MetricRegistry/name ^String @group ^"[Ljava.lang.String;" (into-array String [category metric]))
-        tagged-metric (.tagged ^MetricName metric-name ^"[Ljava.lang.String;" (into-array String ["actor" @group]))]
-    (.histogram ^MetricRegistry metrics-registry tagged-metric)))
+  ([category metric]
+   (mk-histogram category metric nil))
+  ([category metric topic-entity-name]
+   (let [metric-name   (MetricRegistry/name ^String @group ^"[Ljava.lang.String;" (into-array String [category metric]))
+         tags          (if (nil? topic-entity-name)
+                         ["actor" @group]
+                         ["actor" @group "topic_name" topic-entity-name])
+         tagged-metric (.tagged ^MetricName metric-name ^"[Ljava.lang.String;" (into-array String tags))]
+     (.histogram ^MetricRegistry metrics-registry tagged-metric))))
 
 (defn increment-count
   ([metric-namespace metric]
-   (increment-count metric-namespace metric 1))
-  ([metric-namespace metric n]
-   (let [meter ^Meter (mk-meter metric-namespace metric)]
+   (increment-count metric-namespace metric nil 1))
+  ([metric-namespace metric topic-entity-name]
+   (increment-count metric-namespace metric topic-entity-name 1))
+  ([metric-namespace metric topic-entity-name n]
+   (let [meter ^Meter (mk-meter metric-namespace metric topic-entity-name)]
      (.mark meter (int n)))))
 
 (defn decrement-count
   ([metric-namespace metric]
-   (decrement-count metric-namespace metric 1))
-  ([metric-namespace metric n]
-   (let [meter ^Meter (mk-meter metric-namespace metric)]
+   (decrement-count metric-namespace metric nil 1))
+  ([metric-namespace metric topic-entity-name]
+   (decrement-count metric-namespace metric topic-entity-name 1))
+  ([metric-namespace metric topic-entity-name n]
```

Comment:
The case of [metric-namespace metric n] is not handled here

Reply:
same as above..

Reply:
done, removed it